### PR TITLE
docs: errors

### PR DESCRIPTION
**Pull request by odedhutzler**
### Description of the Changes

Error documentation

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

**Comment by dan-ziv** [comment]
@odedhutzler Why this is an html content and not md???

File: docs/error.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Error
+
+An object representing an error that can be triggered / thrown.
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: An error is defined as an object that represents an error that can be triggered or thrown.

File: docs/error.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+## **Overview**
+
+The errors are used to notify the user of a critical issue or log a recoverable one. An error object is consisted of three mandatory parameters: error severity, error category and an error code. The forth parameter is a mandatory object with more information about the issue.
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: Errors are used to notify the user of a critical issue or to log information on a recoverable problem. An error object consists of three mandatory parameters: error severity, error category and an error code. The forth parameter is a mandatory object with additional information about the issue.

File: docs/error.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+There are two severities to an error:
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: There are two error severity levels: Critical and recoverable.

File: docs/error.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+critical and recoverable.
+
+A critical error is triggered in an error event, and is shown to the user as error overlay in the player itself.
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: and is shown to the user as an error overlay in the player itself

File: docs/error.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+A recoverable error is not shown to an end user, but appears in the console.
+
+---------
+
+## **Creating an Error object**
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: If this is a heading you don't need the bold!

## Creating an Error Object

File: docs/error.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+ Import the error class:
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: Step 1: Import the error class:

File: docs/error.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+```javascript
+import Error from 'path/to/error/class/../error'
+```
+
+
+Creating an error object:
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: Step 2: Create an error object:

File: docs/error.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+```javascript
+
+new Error(Error.Severity.CRITICAL,
+		    Error.Category.NETWORK,
+		    Error.Code.HTTP_ERROR,
+		     {'url': 'www.some-bad-url.com'})
+```
+
+## **Error string**
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: ## Error String

File: docs/error.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+Browser console string: (will be present when critical & recoverable)
+
+[Error] Player error NETWORK.HTTP_ERROR 'url':'www.some-bad-url.com'
+
+## **Severity list:**
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: ## Severity List

File: docs/error.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
...colgroup><tr><th style="text-align: left;" class="confluenceTh">Name</th><th style="text-align: left;" class="confluenceTh">Type</th><th class="last confluenceTh" style="text-align: left;">Description</th></tr></thead><tbody><tr><td class="name confluenceTd" style="text-align: left;"><code>RECOVERABLE</code></td><td class="type confluenceTd" style="text-align: left;"><span class="param-type">1</span></td><td class="description last confluenceTd" style="text-align: left;">An error occurred, but the Player is attempting to recover from the error. If the Player cannot ultimately recover, it still may not throw a CRITICAL error. For example, retrying for a media segment will never result in a CRITICAL error (the Player will just retry forever).</td></tr><tr><td class="name confluenceTd" style="text-align: left;"><code>CRITICAL</code></td><td class="type confluenceTd" style="text-align: left;"><span class="param-type">2</span></td><td class="description last confluenceTd" style="text-align: left;">A critical error that the library cannot recover from. These usually cause the Player to stop loading or updating. A new manifest must be loaded to reset the library.</td></tr></tbody></table>
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: we used player lowercase everywhere else so please change to player

File: docs/error.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
+
+## **Categories List:**
```
Comment (AnnaAleksandrowicz) [inline docs/error.md]: ## Categories List
Errors fall under the following categories:

[see if this addition is OK with you - there is not info yet on categories so i'm guessing...]

File: docs/error.md (hunk 3 above)
Comment (OrenMe) [reply docs/error.md]: * Critical
* Recoverable

File: docs/error.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+		    Error.Category.NETWORK,
+		    Error.Code.HTTP_ERROR,
+		     {'url': 'www.some-bad-url.com'})
+```
+
+## **Error string**
+
+Browser console string: (will be present when critical & recoverable)
+
+[Error] Player error NETWORK.HTTP_ERROR 'url':'www.some-bad-url.com'
```
Comment (OrenMe) [inline docs/error.md]: can you please explain what you are trying to show in this section?

File: docs/error.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+ player.addEventListener(player.Event.ERROR, e => {
+	 const error = e.payload;
+	 const severity = error.severity;
+	 const category = error.category;
+	 const code = error.code;
+	 const moreInfo = error.data;
+ })
+```
+
+## **Error string**
```
Comment (OrenMe) [inline docs/error.md]: I don't think this section and title make much sense.
I understand what you are trying to say but don't think we are saying it well enough.
Need to make it more like - if you want to debug player and see explicit error messages then you need to use debug mode which will enable seeing explicit error messages in console.
This is just general, not actual words I will use.
you can consult with @AnnaAleksandrowicz if you don't have a better way to express this.

Also, I would connect this to the debug article we are adding, as we are mentioning it here, but it's not clear how to set it in this scope. @dan-ziv FYI.

File: docs/error.md [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+
+## Registering to an error event
+```javascript
+ player.addEventListener(player.Event.ERROR, e => {
+	 const error = e.payload;
+	 const severity = error.severity;
+	 const category = error.category;
+	 const code = error.code;
+	 const moreInfo = error.data;
```
Comment (OrenMe) [inline docs/error.md]: Maybe add to code snippet something like:
```
const error = e.payload;
console.error(
	 error.severity,
	 error.category,
	 error.code,
	 error.data
});
```

File: docs/error.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+---------
+
+## **Creating an Error object**
+
+
+Creating an error object:
+
+```javascript
+
+new Error(Error.Severity.CRITICAL,
```
Comment (OrenMe) [inline docs/error.md]: align code snippet

File: docs/error.md [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+## **Creating an Error object**
+
+
+Creating an error object:
+
+```javascript
+new Error(Error.Severity.CRITICAL,
+    Error.Category.NETWORK,
+    Error.Code.HTTP_ERROR,
+    {'url': 'www.some-bad-url.com'})
```
Comment (dan-ziv) [inline docs/error.md]: Missing semicolon

File: docs/error.md [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+
+---------
+
+## **Creating an Error object**
+
+
+Creating an error object:
+
+```javascript
+new Error(Error.Severity.CRITICAL,
```
Comment (dan-ziv) [inline docs/error.md]: What they should or can do with the error they just created? need to add explanation.

File: docs/error.md [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+## **Using Debug Mode**
+
+To debug the player and view explicit error messages in the console, you'll need to use debug mode.
+
+In order to run the player in debug mode, add '?debugKalturaPlayer' to the URI. More information about debugging and troubleshooting of the player can be found [here](here).
+
+An example to an error message:
+[Error] Player error NETWORK.HTTP_ERROR 'url':'www.some-bad-url.com'
```
Comment (dan-ziv) [inline docs/error.md]: When you are describing some output of the console wrap it with code so it will be understood.

File: docs/error.md (hunk 16 above)
Comment (OrenMe) [reply docs/error.md]: @odedhutzler ?

File: docs/error.md [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+
+```javascript
+player.addEventListener(player.Event.ERROR, e => {
+  const error = e.payload;
+  console.error(
+	error.severity,
+   	error.category,
+   	error.code,
+   	error.data);
+})
```
Comment (OrenMe) [inline docs/error.md]: missing semicolon

File: docs/error.md [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+
+
+
+## **Using Debug Mode to see Explicit Error Messeges**
+
+To debug the player and view explicit error messages in the console, you'll need to use debug mode.
+
+In order to run the player in debug mode, add '?debugKalturaPlayer' to the URI.
+
+> More information about debugging and troubleshooting of the player can be found [here](./debug.md).
```
Comment (OrenMe) [inline docs/error.md]: link is not available - debug is on the kaltura-player repo.
also `?debugKalturaPlayer` is a feature of kaltura-player, so better set the scope to playkit here - just fix the link above to mention that debug mode is set via config.debugLevel

**Comment by dan-ziv** [comment]
Moving this to kaltura player repo.

**Pull request closed without merging**